src/hotspot/share/cds/filemap.cpp

```diff
@@ -1889,7 +1889,7 @@ MapArchiveResult FileMapInfo::map_region(int i, intx addr_delta, char* mapped_ba
   FileMapRegion* r = region_at(i);
   size_t size = r->used_aligned();
   char *requested_addr = mapped_base_address + r->mapping_offset();
-  assert(r->mapped_base() == nullptr, "must be not mapped yet");
+  assert(!is_mapped(), "must be not mapped yet");
   assert(requested_addr != nullptr, "must be specified");
 
   r->set_mapped_from_file(false);
```

Comment:
Need to add `r->set_in_reserved_space(false);` here, as the `r` can be used twice (first map at requested location; if that fails, map at random location).

Reply:
`FileMapRegion::init()` is called once, during dump time. `r->set_in_reserved_space(false)` should be called there as well.

During run time, `FileMapInfo::map_region()` can be called twice, on the same index `i`, so the `_in_reserved_space` needs to be reset at the beginning of each call. That would be similar to what we do with `_mapped_from_file`.